### PR TITLE
Limit amount of retries while generating unique codes

### LINKED ISSUE #74: [API] Limit the amount of retries when generating codes
The API currently retries a code generation unlimited times until it is unique. This should not be the case. We need to limit the amount of times it retries the insertion, and send an error response when that happens, prompting the user to retry the request if needed.

### PR DESCRIPTION
**Pull request by TiagoMMDavid**
This PR limits the possible amount of retries the server application does while trying to generate unique codes. If the limit is exceeded, an exception is thrown.

Closes GH-74.

### CODE REVIEW

File: code/jvm/src/main/kotlin/org/ionproject/codegarten/pipeline/exceptionhandling/ExceptionHandler.kt [hunk 1]
```diff
@@ -104,6 +105,19 @@ class ExceptionHandler {
...
 
+    @ExceptionHandler(value = [LoopDetectedException::class])
+    private fun handleForbiddenException(
+        ex: LoopDetectedException,
+        request: HttpServletRequest
+    ): ResponseEntity<Response> =
+        handleExceptionResponse(
+            URI("/problems/loop-detected").includeHost(),
+            "Loop Detected",
+            HttpStatus.LOOP_DETECTED,
```
Comment (dvsshadow) [inline code/jvm/src/main/kotlin/org/ionproject/codegarten/pipeline/exceptionhandling/ExceptionHandler.kt]: Shouldn't it just be status code 500?
The [508](https://developer.cdn.mozilla.net/en-US/docs/Web/HTTP/Status/508) status code specficiation does not seem like the best option.

**Review by dvsshadow — CHANGES_REQUESTED** [review]
Left a comment.

**Review by PTKay — APPROVED** [review]
LGTM

**Review by dvsshadow — APPROVED** [review]
👍

**Pull request merged**